src/proposed.api.d.ts

```diff
@@ -0,0 +1,131 @@
+     * are in the above local directory.
+     *
+     * Using this setting one can map the local directory to the remote directory.
+     * In this case the value of this property would be /users/xyz/remoteServer.
+     *
+     * Note: A side effect of providing this value is the session names are generated the way they are in Jupyter Notebook/Lab.
+     * I.e. the session names map to the relative path of the notebook file.
+     * As a result when attempting to create a new session for a notebook/file, Jupyter will
+     * first check if a session already exists for the same file and same kernel, and if so, will re-use that session.
+     */
+    readonly mappedRemoteNotebookDir?: string;
+    /**
+     * Returns the sub-protocols to be used. See details of `protocols` here https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket
+     * Useful if there is a custom authentication scheme that needs to be used for WebSocket connections.
+     * Note: The client side npm package @jupyterlab/services uses WebSockets to connect to remote Kernels.
+     */
+    readonly webSocketProtocols?: string[];
+}
+
+export interface JupyterServer extends Disposable {
+    readonly id: string;
+    /**
+     * A human-readable string which is rendered prominent.
+     */
+    label: string;
+    resolveConnectionInformation: () => Promise<JupyterServerConnectionInformation>;
+}
+
+export class JupyterServerCreationItem extends Disposable {
+    /**
+     * A human-readable string which is rendered prominent. Supports rendering of {@link ThemeIcon theme icons} via
+     * the `$(<name>)`-syntax.
+     */
+    label: string;
+    /**
+     * A human-readable string which is rendered less prominent in a separate line. Supports rendering of
+     * {@link ThemeIcon theme icons} via the `$(<name>)`-syntax.
+     */
+    detail?: string;
+    /**
+     * A string that should be used when comparing this item
+     * with other items. When `falsy` the {@link JupyterServerCreationItem.label label}
+     * is used.
+     */
+    sortText?: string;
+    /**
+     * Optional flag indicating if this item is picked by default.
+     * If there are no existing servers, and this flag is true, then this item will be picked by default.
+     *
+     * Note: this property is ignored when {@link JupyterServerCollection.createServer createJupyterServer} has been called.
+     */
+    picked?: boolean;
+}
+
+export class JupyterServerCollection extends Disposable {
+    /**
+     * Identifier must be globally unique.
+     */
+    readonly id: string;
+    /**
+     * A human-readable string which is rendered prominent.
+     */
+    label: string;
+    /**
+     * A link to a page providing more information to the user about this item.
+     */
+    documentation?: Uri;
+    /**
+     * Creates an entry in the list of Jupyter Servers from which the user can pick.
+     *
+     * @param {string} id
```

Comment:
[nit] Don't use types in JSDoc if you're writing typescript

Reply:
You also don't need `memeberof`

Reply:
I don't manually type these, just use an extension that auto populates them.
will remove the types in the final *.d.ts file